app/models/application_record.rb

```diff
@@ -2,9 +2,11 @@ class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
 
   def append_url_options(path, options = {})
-    if options[:format] && options[:locale]
+    locale = normalise_locale(options[:locale])
+
+    if options[:format] && options[:locale] && locale != I18n.default_locale
```

Comment:
Nice find!  I'd missed the possibility of `:en` being passed in as a parameter, so this should make the code more robust.